src/Compiler.Dynamic/SystemWebCompilation.cs

```diff
@@ -188,11 +195,52 @@ private async Task<T> InternalCompilePageAsync(string path, CancellationToken to
         }
     }
 
-    private async Task<T[]> GetDependencyPages(TemplateParser parser, CancellationToken token)
+    private static List<string> GetAllDependencies(string path)
+    {
+        var pageDependencyParser = new PageDependencyParser();
```

Comment:
why this implementation? Does overriding the virtual methods in the UserContorlParser not work?

Reply:
Looks like you may want to override `TemplateParser`

```cs
    internal virtual IEnumerable<string> GetDependencyPaths()
    {
        foreach (var t in TagRegisterEntries)
        {
            yield return t.VirtualPath;
        }
    }
```